protos/sources/data-node/api/v2/trading_data.proto

```diff
@@ -3316,3 +3322,31 @@ message PingRequest {
 // Ping response from the data node
 message PingResponse {
 }
+
+message EstimatePositionRequest {
+    string market_id = 1;
+    int64 open_volume = 2;
+    repeated vega.Order active_orders = 3;
+    string collateral_available = 4; 
+}
+
+message EstimatePositionResponse {
+    MarginEstimate margin_estimate = 1;
+    LiquidationEstimate liquidation_price_estimate = 2;
+}
+
+message MarginEstimate {
+    vega.MarginLevels worst_case = 1;
+    vega.MarginLevels best_case = 2;
+}
+
+message LiquidationEstimate {
+    LiquidationPrice worst_case = 1;
+    LiquidationPrice best_case = 2;
+}
+
+message LiquidationPrice {
+    string position_only = 1;
+    string including_buy_orders = 2;
+    string including_sell_orders = 3;
+}
```

Comment:
All fields needs to be documented with a doc commented (ended with a `.` so the documentation is generated properly.)